### PR TITLE
Add Telugu News Classification

### PR DESCRIPTION
**Pull request by SaitejaUtpala**
## Checklist for adding MMTEB dataset

Reason for dataset addition: **This is first telugu classification in the mteb**

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: docs/mmteb/points/483.jsonl [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{"GitHub": "SaitejaUtpala", "New dataset": 6}
```
Comment (isaac-chung) [inline docs/mmteb/points/483.jsonl:1]: Telugu has been covered in classification in `MassiveIntentClassification`.
```suggestion
{"GitHub": "SaitejaUtpala", "New dataset": 2}
```

File: mteb/tasks/Classification/tel/TeluguAndhraJyotiNewsClassification.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        bibtex_citation=None,
+        n_samples={"train": 17312, "test": TEST_SAMPLES},
+        avg_character_length={"train": 1435.53, "test": 1428.28},
+    )
+
+    def dataset_transform(self):
+        seed = 42
+        random.seed(seed)
+        self.dataset = self.dataset.rename_column("body", "text")
+        self.dataset = self.dataset.rename_column("topic", "label")
```
Comment (isaac-chung) [inline mteb/tasks/Classification/tel/TeluguAndhraJyotiNewsClassification.py:44]:
```suggestion
        self.dataset = self.dataset.rename_column("body", "text").rename_column("topic", "label")
```

File: mteb/tasks/Classification/tel/TeluguAndhraJyotiNewsClassification.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+        for split in ["test"]:
+            ds = self.dataset[split]
+            # Determine number of classes and samples per class
+            class_count = Counter([sample["label"] for sample in ds])
+            num_classes = len(class_count)
+            total_samples = min(TEST_SAMPLES, len(ds))
+            samples_per_class = total_samples // num_classes
+
+            # Try to maintain class balance
```
Comment (isaac-chung) [inline mteb/tasks/Classification/tel/TeluguAndhraJyotiNewsClassification.py:54]: this is great! I wonder if something like could help:
```
self.dataset = self.dataset["test"].train_test_split(
            test_size=TEST_SAMPLES, seed=42, stratify_by_column="label"
   ) 
```

**Review by isaac-chung — CHANGES_REQUESTED** [review]
Thanks for your contributions! Just a few comments.

File: mteb/tasks/Classification/tel/TeluguAndhraJyotiNewsClassification.py (hunk 3 above)
Comment (SaitejaUtpala) [reply mteb/tasks/Classification/tel/TeluguAndhraJyotiNewsClassification.py:54]: @isaac-chung  I don't think this is stratified sampling !. This is completely different.  label with least examples is maxed out, so that class imbalance is minimized

Comment (isaac-chung) [reply mteb/tasks/Classification/tel/TeluguAndhraJyotiNewsClassification.py:54]: I see, thanks!. @imenelydiaker and I are actually going through past datasets and we saw this as well in `VieStudentFeedbackClassification`. I wonder if we'd like to keep our stratified sampling strategy as default for all (to me this is sufficient), or adopt this strategy here.

Comment (imenelydiaker) [reply mteb/tasks/Classification/tel/TeluguAndhraJyotiNewsClassification.py:54]: The stratification process is just here because of the subsampling, it allows us to keep the same samples a'd labels proportions as in the original dataset. Note that without the subsampling we would have run the evaluation on the whole dataset even if it's imbalanced.

So I think the question we should be asking is: do we want to have imbalanced classes during the evaluation? Because we're using KNN to evaluate the classification task, after that we compute an accuracy whatever the class balance is.

Comment (isaac-chung) [reply mteb/tasks/Classification/tel/TeluguAndhraJyotiNewsClassification.py:54]: Make sense. Having imbalanced classes are realistic and IMO inevitable. The main score for those datasets could be f1 instead of accuracy. Wdyt?

Comment (imenelydiaker) [reply mteb/tasks/Classification/tel/TeluguAndhraJyotiNewsClassification.py:54]: Yes I think having f1 as a main score is a good idea! Should we update previous tasks that used accuracy?

Comment (isaac-chung) [reply mteb/tasks/Classification/tel/TeluguAndhraJyotiNewsClassification.py:54]: Yes if those datasets are noticeably imbalanced.
